flink-core/src/main/java/org/apache/flink/api/connector/source/lib/GeneratorSource.java

```diff
@@ -46,25 +46,10 @@
 import static org.apache.flink.util.Preconditions.checkArgument;
 import static org.apache.flink.util.Preconditions.checkNotNull;
 
-/**
- * A data source that produces a sequence of numbers (longs). This source is useful for testing and
- * for cases that just need a stream of N events of any kind.
- *
- * <p>The source splits the sequence into as many parallel sub-sequences as there are parallel
- * source readers. Each sub-sequence will be produced in order. Consequently, if the parallelism is
- * limited to one, this will produce one sequence in order.
- *
- * <p>This source is always bounded. For very long sequences (for example over the entire domain of
- * long integer values), user may want to consider executing the application in a streaming manner,
- * because, despite the fact that the produced stream is bounded, the end bound is pretty far away.
- */
 @Public
-public class NumberSequenceSource
-        implements Source<
-                        Long,
-                        NumberSequenceSource.NumberSequenceSplit,
-                        Collection<NumberSequenceSource.NumberSequenceSplit>>,
-                ResultTypeQueryable<Long> {
+public class GeneratorSource<OUT>
```

Comment:
Just FYI: you need to keep the `NumberSequenceSource` because it is `Public` and can only be changed in a breaking way with a new major release.